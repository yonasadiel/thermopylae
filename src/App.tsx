import { useState } from 'react';
import Terminal from './components/terminal/Terminal';
import Quote from './components/quote/Quote';
import Settings from './components/settings/Settings';
import Particles from './components/particles/Particles';
import { useSettings } from './hooks';
import './App.css';

const App = () => {
    const [openSettings, setOpenSettings] = useState<boolean>(false);
    const { settings } = useSettings();
    return (
        <div
            className="main"
            style={{
                "--background-color": settings.themeBackgroundColor,
                "--foreground-color": settings.themeForegroundColor,
            } as React.CSSProperties}
        >
            <Particles />
            <img src={settings.themeBackgroundImageBase64 ? `data:image/jpeg;base64,` +  settings.themeBackgroundImageBase64 : ""} />
            <div className="center">
                <Terminal />
                <Quote />
            </div>
            <div className="top-right">
                <button type="button" className={`settings-button ${openSettings ? '' : 'show'}`} onClick={() => setOpenSettings(true)}>
                    {/* TODO: make this custom color */}
                    <img alt="settings" src="./assets/gear.svg" width="20px" height="20px" />
                </button>
            </div>
            <div className={`settings-mask ${openSettings ? 'active' : ''}`}>
                {(<Settings onClose={() => setOpenSettings(false)} />)}
            </div>
        </div >
    );
};

export default App;
