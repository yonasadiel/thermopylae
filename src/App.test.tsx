import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import App from './App';
import { fn } from 'jest-mock';


describe('App', () => {
    test('should render correctly', () => {
        window.particlesJS = fn();
        const result = render(<App />);
        expect(result.baseElement).toMatchSnapshot();
    });
});
