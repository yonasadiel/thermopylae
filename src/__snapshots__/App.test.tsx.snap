// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<body>
  <div>
    <div
      class="main"
      style="--background-color: black; --foreground-color: white;"
    >
      <div
        id="particles-js"
        style="display: block; z-index: 0;"
      />
      <img />
      <div
        class="center"
      >
        <div
          class="terminal"
        >
          <input
            list="bangs"
            type="text"
            value=""
          />
          <div
            id="bangs"
          />
        </div>
        <div
          class="quote"
          style="display: flex;"
        >
          <div
            class="text-line"
          >
            <span>
              Defensive Programming
            </span>
          </div>
          <div
            class="subtext-line"
          >
            <span>
              Defensive programming is a form of defensive design intended to develop programs that are capable of detecting potential security abnormalities and make predetermined responses. It ensures the continuing function of a piece of software under unforeseen circumstances. Defensive programming practices are often used where high availability, safety, or security is needed.
            </span>
          </div>
        </div>
      </div>
      <div
        class="top-right"
      >
        <button
          class="settings-button show"
          type="button"
        >
          <img
            alt="settings"
            height="20px"
            src="./assets/gear.svg"
            width="20px"
          />
        </button>
      </div>
      <div
        class="settings-mask "
      >
        <div
          class="settings"
        >
          <div
            class="close-button"
          >
            <img
              alt="cross"
              height="20px"
              src="/assets/cross.svg"
              width="20px"
            />
          </div>
          <div
            class="navbar"
          >
            <div
              class="menu active"
            >
              Theme
            </div>
            <div
              class="menu "
            >
              Quote
            </div>
            <div
              class="menu "
            >
              Terminal
            </div>
          </div>
          <div
            class="divider"
          />
          <div
            class="settings-content"
          >
            <div
              class="theme-settings"
            >
              <div
                class="input-group"
              >
                <label>
                  Style
                </label>
                <p>
                  <input
                    type="color"
                    value="black"
                  />
                  <span>
                    Background Color
                  </span>
                </p>
                <p>
                  <input
                    type="color"
                    value="white"
                  />
                  <span>
                    Color
                  </span>
                </p>
              </div>
              <div
                class="input-group"
              >
                <label>
                  Background Image
                </label>
                <input
                  type="file"
                  value=""
                />
              </div>
              <div
                class="input-group"
              >
                <label
                  for="particles-enabled"
                >
                  Particles
                </label>
                <input
                  checked=""
                  id="particles-enabled"
                  type="checkbox"
                />
                <span>
                  Enabled
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
